### ISSUE TITLE
Deserialize derived type giving base type

### ISSUE THREAD

**Issue by amedeomar** [QuickenLoans/XSerializer#40]
Hi,
is it possibile to deserialize a class knowing its base class?

Example:

IXmlSerializer serializer = XmlSerializer.Create(typeof(MyBaseClass));
var d = (MyDerivedClass)serializer.Deserialize(stream);

**Comment by bfriesen** [comment]
Yes, you can deserialize a class as its base class (or interface). Two things must occur for this to work. First, a "type hint" needs to be provided in the xml. (`xsi:type="MyDerivedClass"`, in the example below).

```xml
<MyBaseClass xmlns:xsd="http://www.w3.org/2001/XMLSchema"
    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
    xsi:type="MyDerivedClass">
</MyBaseClass>
```

Second, the expected derived types must be supplied to the serializer. This can be done in two ways. Either pass in the type to the constructor/factory method:

```c#
IXmlSerializer serializer = XmlSerializer.Create(typeof(MyBaseClass), typeof(MyDerivedClass));
```

Or decorate the base class with an `[XmlInclude]` attribute:

```c#
[XmlInclude(typeof(MyDerivedClass))]
public class MyBaseClass
{
}
```

Note: Use the same technique to serialize objects as their base type in order to include all of the derived type's properties.

**Comment by bfriesen** [comment]
@amedeomar Does this answer your question?

**Comment by amedeomar** [comment]
Hi @bfriesen sorry for the late reply, yes your answer is quite clear, the problem was that I serialized with:
IXmlSerializer serializer = XmlSerializer.Create(typeof(MyDericedClass));
serializer.Serialize(...)

and then i deserialized with

IXmlSerializer serializer = XmlSerializer.Create(typeof(MyBaseClass));
serializer.Deserialize(...)

this way the "type hint" there was not in the xml and the deserailization did not work :D
